src/env.cc

```diff
@@ -1051,6 +1052,25 @@ void Environment::AddArrayBufferAllocatorToKeepAliveUntilIsolateDispose(
   keep_alive_allocators_->insert(allocator);
 }
 
+bool Environment::RunWeakRefCleanup() {
+  isolate()->ClearKeptObjects();
+
+  TryCatchScope try_catch(this);
+  while (!cleanup_finalization_groups_.empty()) {
+    Local<FinalizationGroup> fg =
+        cleanup_finalization_groups_.front().Get(isolate());
```

Comment:
Does it that matter that this executes the cleanups in LIFO order? Looking at `Heap::GarbageCollectionEpilogue()` I think the answer is 'no' but I thought I'd ask anyway.